src/containers/forms/PractitionerForm/UserIdSelect/index.tsx

```diff
@@ -0,0 +1,83 @@
+/** wrapper around react-select component that enables one
+ * to select a single user from openmrs
+ */
+import React, { useEffect, useState } from 'react';
+import Select from 'react-select';
+import { ValueType } from 'react-select/src/types';
+import { OpenSRPService } from '../../../../services/opensrp';
+
+const OPENMRS_USERS_REQUEST_PAGE_SIZE = 100;
+
+// props interface to UserIdSelect component
+export interface Props {
+  onChangeHandler?: (value: Option) => void;
+  serviceClass: typeof OpenSRPService;
+}
+
+export const defaultProps = {
+  serviceClass: OpenSRPService,
+};
+
+export interface Option {
+  label: string;
+  value: string;
+}
+
+/** interface describing a user from openmrs */
+interface OpenMRSUser {
+  id: string;
+}
+
+/** The UserIdSelect component */
+export const UserIdSelect: React.FC<Props> = props => {
+  const { onChangeHandler: onChange } = props;
+  const [openMRSUsers, setOpenMRSUsers] = useState<OpenMRSUser[]>([]);
+
+  /** calls the prop.onChange with only the userId
+   * @param {ValueType<Option>} option - the value in the react-select
+   */
+  const changeHandler = (option: ValueType<Option>) => {
+    if (option !== null && option !== undefined && onChange) {
+      onChange(option as Option);
+    }
+  };
+
+  /** Pulls all openMRS users info and puts in store */
+  const loadOpenMRSUsers = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const currentUserIndex = 0;
+    let filterParams = {
+      page_size: OPENMRS_USERS_REQUEST_PAGE_SIZE,
+      start_index: currentUserIndex,
+    };
+    const serve = new service('user');
+    let responseSize: number = 100;
+
+    while (responseSize === OPENMRS_USERS_REQUEST_PAGE_SIZE) {
```

Comment:
Questionable: Each time the component mounts, it makes iterative calls to opensrp untill it gets all users from openSRP proxied via opensrp.

Reply:
What happens here if there are exactly 100 openMRS users to return? Oh the second call will have zero results, which should have no effect when calling `setOpenMRSUsers` with an empty array, ama?

Reply:
oops there is a bug, each call to setState will override already stored openmrs users, ill fix that quckly. 
@cKellyDesign The logic is that each request will have results array with a max length  of a 100 objects. So if we get less that a 100 records in any request then that's how i know that the page returned in the that request is the last page, and therefore i should not make any subsequent calls.
so ``` while any request returns the maximum request size:.
assume there is more data and make another request,
else know we just got the last page and stop making further requests ```